cpp/src/strings/regex/regcomp.cpp

```diff
@@ -439,15 +445,70 @@ class regex_parser
                     return BOL;
                 case 'Z':
                     return EOL;
+                default:
+                {
+                    // let valid escapable chars fall through as literal CHAR
+                    if( yy && (escapable_chars.find(static_cast<char>(yy)) != escapable_chars.end()) )
+                        break;
+                    // anything else is a bad escape so throw an error
+                    std::ostringstream message;
```

Comment:
Looks like we need a CUDF_FAIL that takes a non-literal string.

Reply:
I don't think the dynamic string is really necessary here. It would be perfectly fine to just say:
```
CUDF_FAIL("Invalid regex pattern: bad escape");`
```

There aren't other places in libcudf where we try and further introspect the error condition. E.g., if checking for if two columns have the same type, we just say "type mismatch". We don't try and introspect what the two different types are and print them.

Reply:
The error is reporting where in the regex pattern the offending character was found. I've seen some really messed up regex patterns. This is helpful in telling the user (not necessarily a coder) where the bad character was found in very long, complicate expressions.
For example, where is the bad escape sequence in the following?
```
"\"\a\b\\%\p\t\7\\\b\A\Z]\"(?:|(?:.*?(?![^\\]\\\")).?)[^\\]?\",?\s*)*)\]"
```
We have the information and I don't see why we cannot report this to the user.

Reply:
You now have to maintain the extra logic of detecting the "what" and "where" of the error, which adds extra complexity (as evidenced by the fact that `CUDF_FAIL` isn't being used directly and instead manually throwing an exception). 

This may not seem like a big deal isolated to this feature, but imagine multiplying this out to every exception thrown in libcudf. 

It also sets up a dangerous precedent where the user might expect libcudf to do all of the work of trying to detect all possible bad regex patterns and report exactly what is wrong with them. 

> We have the information and I don't see why we cannot report this to the user.

The same is true for every exception thrown in libcudf. Just because you _can_ do something, doesn't mean you _should_. 

If you're not convinced and do decide to keep it, don't use `ostringstream`. It's way slower than just doing regular old string concatenation. Also, add a `CUDF_FAIL_STRING` macro that abstracts the throwing of `cudf::logic_error` with a non string literal.